File: programs/Ds/Stack.java
import java.util.*;
class Stack
{
    static Scanner sc=new Scanner(System.in);
    static int stack[],top=-1,pop,size;
    static
    {
Stack.create();
    }
    public static void main(String args[])
    
    {
        int ch,item;
        
        while(true)
        {
            System.out.println("1.push");
        System.out.println("2.pop");
        System.out.println("3.peek");
        System.out.println("4.traverse");
        System.out.println("5.Quit");

        System.out.println("Enter your  choice:");
        ch=Stack.sc.nextInt();
        switch(ch)
        {
            case 1:System.out.println("Enter element:");
            item=sc.nextInt();
           Stack.push(item);
            break;
            case 2:item=Stack.pop();
            if(item==0)
            {
                System.out.println("Stack is underflow\n");
            }
            else
            {
                System.out.println("popped item:"+item);
            }
            break;
            case 3:peek();
            item=Stack.pop();
            if(item==0)
            {
                System.out.println("Stack is underflow\n");
            }
            else
            {
                System.out.println("peeked item:"+item);
            }
            break;
            case 4:Stack.traverse();
            break;
            case 5:System.exit(1);
            default:System.out.println("invalid choice:");
        }
        }
    }
    static void create()
    {

System.out.println("Enter size of stack:");
      size=Stack.sc.nextInt();
     Stack.stack=new int [size];
     System.out.println("Stack created with size:"+size);
    }
    static void push(int item)
    {
        if(Stack.isFull())
        {
System.out.println("Stack is overflow:\n");
        }
        else
        {
     Stack.stack[++top]=item;
        }
    }

static boolean isFull()
{
    if(top==size-1)
{
return true;
}
else
{
return false;
}
}

static int pop()
{
    if(Stack.isEmpty())
    {
return 0;    }
    else
    {
        return stack[top--];
    }
}

static boolean isEmpty()
{
    if(top==-1)
    {
return true;
    }
    else
    {
        return false;
    }
}

static int peek()
{
    if(Stack.isEmpty())
    {
return 0;    
}
    else
    {
        return stack[top];
    }
}

static void traverse()
{
    if(Stack.isEmpty())
    {
        System.out.println("stack is empty\n");
    }
    else
    {
        System.out.println("Elements of stack:");
        for(int i=top;i>=0;i--)
        {
            System.out.println(stack[i]);
        }
    }
}
}

File: programs/Ds/Stack.c
 /* Stack implementation using static array */
#include<stdio.h>
#include<stdlib.h>
#define CAPACITY 5//pre-processor micro
int stack[CAPACITY],top=-1,item;
void push(int);
int pop(void);
int isFull(void);
int isEmpty(void);
void traverse(void);
void peek(void);
void main()
{
    int ch;
    while(1)
{
    printf("1.push \n");
     printf("2.pop \n");
   printf("3.peek \n");
     printf("4.Traverse \n");
   printf("5.Quit \n");

     printf("Enter your choice:");
     scanf("%d",&ch);

     switch(ch)
     {
         case 1:printf("Enter element :");
         scanf("%d",&item);
         push(item);
         break;
         case 2:item=pop();
         if(item==0)
         {
             printf("stack is underflow\n");
         }
             else
             {
                 printf("poped item:%d\n",item);
             }
         
         break;
         case 3:peek();
            break;
         case 4:traverse();
         break;
         case 5:exit(0);
         default:
         printf("invalid input \n \n");
     }
     }
}
    


void push(int ele )
{
    if(isFull())
    {
printf("stack is overflow\n");
    }
    else
    {
top++;
 stack[top]=ele;
 printf("%d pushed \n",ele);
    }
}
int isFull()
{
    if(top==CAPACITY-1)
    {
return 1;
    }
    else
    {
return 0;
    }
}

int pop()
{
    if(isEmpty())
    {
        
        return 0;
    }
    else
    {
        return stack[top--];
        
    }
}

int isEmpty()
{
    if(top==-1)
    {
        return 1;
    }
    else
    {
        return 0;
    }
}

void peek()
{
    if(isEmpty())
    {
        printf("stack is empty\n");

    }
    else
    {
printf("peek element: %d \n",stack[top]);
    }
}

void traverse()
{
    if(isEmpty())
    {
        printf("stack is empty \n");
    }
    else
    {
        int i;
        printf("stack element :\n");
        for(i=0;i<=top;i++)
        {
            printf("%d \n",stack[i]);
        }
    }
}